### ISSUE TITLE
Plots of particle multiplicity

### ISSUE THREAD

**Issue by lawrenceleejr** [lawrenceleejr/FragmentationStudy#3]
* Make a histogram of the number of particles per jet after #1 and #2 are done.
* Then let's also make a 2D histogram of the number of particles (on the y axis) vs the lab frame pT of the jet (on the x axis).
* Then we can take the `.ProfileX()` of that 2D histogram to see the behavior of the mean vs pT.

**Comment by lawrenceleejr** [comment]
So moving forward with this side of it a bit (there's a lot of stuff here, so we'll chip at this slowly):

* At this line, can we change it to not look at PT, but instead look at P? You can calculate P easily as pT*cosh(eta). It's in P that I managed to get things to look really nice in Herwig: https://github.com/lawrenceleejr/FragmentationStudy/blob/495f7ecd1ee9dd11605e836606e4a26881919b12/loop.py#L82
* Can you have `loop.py` also store the profiles into the output ROOT file?
* At this line, let's also make it less than 0.01 GeV. https://github.com/lawrenceleejr/FragmentationStudy/blob/495f7ecd1ee9dd11605e836606e4a26881919b12/loop.py#L77
* And then write a script that will read in the root files (produced by `loop.py`) and make a plot of:

Charged+Neutral particle multiplicity on the y axis. Jet P on the x axis. Four profiles drawn. 
* Pythia: Jets in events with a high momentum photon
* Pythia: Jets in three jet events
* Herwig: Jets in events with a high momentum photon
* Herwig: Jets in three jet events

All of this using 1 TeV ee collisions. (I think the second one requires making a new jjj sample at 1 TeV.)

When filling up the histograms, we will eventually want to exclude jets that originate from a gluon as well. That will require asking if the jet has a small deltaR to a gluon that has parents that are the electron and positron.

Extra points if the drawing is done in matplotlib. (You might want to look into using `uproot` to read in the root files for then manipulation in python with matplotlib.) That's what we'll want eventually for a final thing.

**Comment by lawrenceleejr** [comment]
One potential way of showing a comparison of these shapes as a function of momentum is to form so-called split violin plots. 

Can easily use seaborn for that. See https://seaborn.pydata.org/generated/seaborn.violinplot.html

<img width="530" alt="Screen Shot 2023-01-06 at 12 50 47 PM" src="https://user-images.githubusercontent.com/7438278/211069854-6443269c-5a16-4ccb-8c1b-a5ca78ececa1.png">

This way, we can actually show the distributions as a function of momentum for both the Z+gluon and the Z+gamma. Also this is probably what we should call those samples...

**Comment by lawrenceleejr** [comment]
To split up the Z+gamma and Z+gluon events in Herwig -- these are the draw commands.

```
Delphes->Draw("GenJet.NCharged+GenJet.NNeutrals:GenJet.PT*cosh(GenJet.Eta)","GenJet.Mass>0.01&&abs(GenJet.Mass)<70&&(MinIf$(GenJet.Mass,GenJet.PT>50)>0.01) ","colz profx ")
Delphes->Draw("GenJet.NCharged+GenJet.NNeutrals:GenJet.PT*cosh(GenJet.Eta)","GenJet.Mass>5&&abs(GenJet.Mass)<70&&(MinIf$(GenJet.Mass,GenJet.PT>50)<0.01) ","colz profx same")
```

First one (Z+gamma):
* Only look at events with a high pT (maybe >20 GeV) photon (as determined by the GenJet mass).
* And then analyze the jets in these events (that are not photons/gluon)

Second one (Z+jet):
* Only look at events that *don't* have a high pT photon
* And then analyze the jets in these events that are not photons/gluon.

**Comment by lawrenceleejr** [comment]
For the split violin plots, @jslawless can help out with some of the tricky data handling that will be needed for that. We can focus on that plot the week of the 16th. So let's focus on getting the root files with the histograms all ready for that during next week.

**Comment by cbell1402** [comment]
Attached are the profiles for eeToZgamma and eeToZgluon plots merged into one. 
![image](https://user-images.githubusercontent.com/93935192/211637434-1c334d3a-e0f2-4213-b424-8723f650d81d.png)

**Comment by lawrenceleejr** [comment]
Very nice. Note to self to discuss: it might be interesting to make the Z+gamma plot where the momentum is in the rest frame of the Z.

**Comment by jslawless** [comment]
I wrote a script that splits a 2d histogram in violin plots. Later I can add functionality to it to compare two different distributions. 

Example:
![Screenshot 2023-01-19 at 3 51 14 PM](https://user-images.githubusercontent.com/116592104/213557215-638f8874-3302-42bd-94f6-e4565ab59453.png)
![Screenshot 2023-01-19 at 3 51 40 PM](https://user-images.githubusercontent.com/116592104/213557240-2b952148-4ef1-4cf9-b2b6-3e721fba1b82.png)

Unfortunately I chose a bad 2d histogram and these all look samey, but it works enough as a starting point.

**Comment by lawrenceleejr** [comment]
Yes this is a great starting point. Hopefully we can get you a few histograms soon to try out. One thing that would be good to get started with is the splitting of them. We'll want one side of each violin to be Z+gamma and the other side to be jjj. And yeah it looks like it's already doing the normalization the way we want it to which is great. 

We'll also need to be careful about how the x axis is translated. i.e. we'll want the original x axis to stay the same (here it looks like its being translated to integer bins) But yeah it's beautiful -- looking forward to it!!
